### PR TITLE
FINERACT-1561: Enhancements for APIs

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

In attention for the ticket [Apache Fineract JIRA ticket FINERACT-1561](https://issues.apache.org/jira/browse/FINERACT-1561).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: .github/workflows/sonarqube.yml [hunk 1]
```diff
@@ -6,9 +6,9 @@ jobs:
     runs-on: ubuntu-20.04
     env:
         TZ: Asia/Kolkata
-        SONAR_ORGANIZATION: ${{ secrets.SONAR_ORGANIZATION }}
-        SONAR_PROJECT_KEY: ${{ secrets.SONAR_PROJECT_KEY }}
-        SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        SONAR_ORGANIZATION: mifos
```
Comment (vidakovic) [inline .github/workflows/sonarqube.yml:9]: This needs to stay untouched

File: .gitignore [hunk 2]
```diff
@@ -21,4 +21,5 @@ bin/
 fineract-provider/out/
 fineract-provider/config/swagger/config.json
 fineract-provider/config/swagger/fineract-input.yaml
-licenses
\ No newline at end of file
+licenses
```
Comment (vidakovic) [inline .gitignore:24]: Same here... not sure if that is OpenMF specific stuff

File: fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountReadPlatformServiceImpl.java [hunk 3]
```diff
@@ -144,7 +144,7 @@ public List<GLAccountData> retrieveAllGLAccounts(final Integer accountClassifica
             boolean firstWhereConditionAdded = false;
             if (accountClassification != null) {
                 sql += " classification_enum like ?";
-                paramaterArray[arrayPos] = accountClassification;
+                paramaterArray[arrayPos] = accountClassification.shortValue();
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountReadPlatformServiceImpl.java:147]: Any specific reason why this needs to be a "short" type? Seems a bit odd...

**Comment by josehernandezfintecheandomx** [comment]
Included in other PR https://github.com/apache/fineract/pull/2279

**Pull request closed without merging**